### ISSUE TITLE
case sensitive instance size

### ISSUE THREAD

**Issue by mcorbin** [exoscale/terraform-provider-exoscale#40]
I have this Terraform configuration:

```
resource "exoscale_compute" "test" {

  display_name = "test-bug-nic"
  template_id  = "c00df33a-df5b-47d2-b9c8-a2eee9ef24a3"

  zone = "ch-gva-2"

  size            = "tiny"
  disk_size       = 20
  key_pair        = "perso"
  security_groups = ["default"]
}

```

Terraform apply:

```
Terraform will perform the following actions:

  # exoscale_compute.test will be created
  + resource "exoscale_compute" "test" {
      + affinity_group_ids = (known after apply)
      + affinity_groups    = (known after apply)
      + disk_size          = 20
      + display_name       = "test-bug-nic"
      + gateway            = (known after apply)
      + id                 = (known after apply)
      + ip4                = true
      + ip6                = false
      + ip6_address        = (known after apply)
      + ip6_cidr           = (known after apply)
      + ip_address         = (known after apply)
      + key_pair           = "perso"
      + name               = (known after apply)
      + password           = (sensitive value)
      + security_group_ids = (known after apply)
      + security_groups    = [
          + "default",
        ]
      + size               = "tiny"
      + state              = (known after apply)
      + tags               = (known after apply)
      + template           = (known after apply)
      + template_id        = "c00df33a-df5b-47d2-b9c8-a2eee9ef24a3"
      + user_data_base64   = (known after apply)
      + username           = (known after apply)
      + zone               = "ch-gva-2"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes
```

If I run `terraform plan` again:

```
Terraform will perform the following actions:

  # exoscale_compute.test will be updated in-place
  ~ resource "exoscale_compute" "test" {
        affinity_group_ids = []
        affinity_groups    = []
        disk_size          = 20
        display_name       = "test-bug-nic"
        gateway            = "185.19.28.1"
        id                 = "ad76a840-d407-4e76-bead-ddda00db347d"
        ip4                = true
        ip6                = false
        ip_address         = "185.19.31.122"
        key_pair           = "perso"
        name               = "test-bug-nic"
        password           = (sensitive value)
        security_group_ids = [
            "f5332702-1fdb-42dd-a68c-fa1008911650",
        ]
        security_groups    = [
            "default",
        ]
      ~ size               = "Tiny" -> "tiny"
        state              = "Running"
        tags               = {}
        template           = "Linux Debian 10 (Buster) 64-bit"
        template_id        = "c00df33a-df5b-47d2-b9c8-a2eee9ef24a3"
        user_data_base64   = true
        username           = "root"
        zone               = "ch-gva-2"
    }

Plan: 0 to add, 1 to change, 0 to destroy.

```

As you can see, configuring the instance `size` with a lowercase offering name will cause Terraform to always update the size. It would be nice to have the size case insensitive if possible.

**Comment by falzm** [comment]
Hi @mcorbin, thank you for reporting this. We'll fix it!

**Comment by ghost** [comment]
This bug no longer exists and can be closed.

**Comment by falzm** [comment]
I still can reproduce it @janoszen.

**Comment by ghost** [comment]
Indeed, my bad.

**Comment by jessicatoscani** [comment]
Fixed for `exoscale_compute_instance` ([v0.33.1](https://github.com/exoscale/terraform-provider-exoscale/blob/v0.33.1/CHANGELOG.md))
